src/daemon/server.rs

```diff
@@ -188,70 +188,108 @@ impl<R: Read, W: Write> Daemon for DaemonServer<R, W> {
         }
     }
 
-    fn color(&self, board: usize) -> Result<Rgb, String> {
-        let path = "/sys/class/leds/system76_acpi::kbd_backlight/color";
-        match fs::read_to_string(&path) {
-            Ok(string) => {
-                let trimmed = string.trim();
-                match Rgb::parse(trimmed) {
-                    Some(color) => Ok(color),
-                    None => Err(format!("Failed to parse keyboard color '{}'", trimmed))
-                }
-            },
-            Err(err) => Err(format!("Failed to read keyboard color: {}", err))
+    fn color(&self, mut board: usize) -> Result<Rgb, String> {
+        #[cfg(target_os = "linux")]
+        {
+            let mut lpc = self.lpc.borrow_mut();
+            if board < lpc.len() {
```

Comment:
This could be something like `if let Some(lpc_board) = lpc.get_mut(board)`, right? Probably slightly more idiomatic that testing length and indexing.

Reply:
True

Reply:
Done in latest commit